pkg/addon/templateagent/values_test.go

```diff
@@ -100,3 +104,155 @@ func TestGetAddOnRegistriesPrivateValuesFromClusterAnnotation(t *testing.T) {
 		})
 	}
 }
+
+func TestGetValues(t *testing.T) {
+	cases := []struct {
+		name             string
+		templateSpec     addonapiv1alpha1.AddOnTemplateSpec
+		values           addonfactory.Values
+		expectedDefault  orderedValues
+		expectedOverride map[string]interface{}
+		expectedPrivate  map[string]interface{}
+	}{
+		{
+			name: "no override",
+			templateSpec: addonapiv1alpha1.AddOnTemplateSpec{
+				Registration: []addonapiv1alpha1.RegistrationSpec{
+					{
+						Type: addonapiv1alpha1.RegistrationTypeKubeClient,
+					},
+				},
+			},
+			expectedDefault: orderedValues{
+				{
+					name:  "HUB_KUBECONFIG",
+					value: "/managed/hub-kubeconfig/kubeconfig",
+				},
+				{
+					name:  "CLUSTER_NAME",
+					value: "test-cluster",
+				},
+			},
+			expectedOverride: map[string]interface{}{
+				"CLUSTER_NAME":   "test-cluster",
+				"HUB_KUBECONFIG": "/managed/hub-kubeconfig/kubeconfig",
+			},
+			expectedPrivate: map[string]interface{}{},
+		},
+		{
+			name:         "overrides builtin value",
+			templateSpec: addonapiv1alpha1.AddOnTemplateSpec{},
+			values: addonfactory.Values{
+				InstallNamespacePrivateValueKey: "default-ns",
+			},
+			expectedDefault: orderedValues{
+				{
+					name:  "CLUSTER_NAME",
+					value: "test-cluster",
+				},
+			},
+			expectedOverride: map[string]interface{}{
+				"CLUSTER_NAME": "test-cluster",
+			},
+			expectedPrivate: map[string]interface{}{
+				InstallNamespacePrivateValueKey: "default-ns",
+			},
+		},
+		{
+			name:         "overrides public value",
+			templateSpec: addonapiv1alpha1.AddOnTemplateSpec{},
+			values: addonfactory.Values{
+				InstallNamespacePrivateValueKey: "default-ns",
+				"key1":                          "value1",
+			},
+			expectedDefault: orderedValues{
+				{
+					name:  "CLUSTER_NAME",
+					value: "test-cluster",
+				},
+			},
+			expectedOverride: map[string]interface{}{
+				"CLUSTER_NAME": "test-cluster",
+				"key1":         "value1",
+			},
+			expectedPrivate: map[string]interface{}{
+				InstallNamespacePrivateValueKey: "default-ns",
+			},
+		},
+		{
+			name:         "default should not be overridden",
```

Comment:
default values can not be overridden, there are some descriptions in the tests that are not precise, I will submit another PR to refactor it after this is merged.